src/HttpClientFactory/Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs

```diff
@@ -47,6 +47,7 @@ public static IServiceCollection AddHttpClient(this IServiceCollection services)
             // Misc infrastructure
             //
             services.TryAddEnumerable(ServiceDescriptor.Singleton<IHttpMessageHandlerBuilderFilter, LoggingHttpMessageHandlerBuilderFilter>());
+            services.TryAddSingleton(new HttpClientMappingRegistry());
```

Comment:
Add a comment here saying that this *has* to be an instance or it'll break the upstream code.